fs-dkr/src/add_party_message.rs

```diff
@@ -49,58 +51,52 @@ pub struct JoinMessage<E: Curve, H: Digest + Clone, const M: usize> {
-    pub(crate) dlog_statement: DLogStatement,
+    pub(crate) dlog_statement: CompositeDLogStatement,
     pub(crate) composite_dlog_proof_base_h1: CompositeDLogProof,
     pub(crate) composite_dlog_proof_base_h2: CompositeDLogProof,
     pub(crate) ring_pedersen_statement: RingPedersenStatement<E, H>,
     pub(crate) ring_pedersen_proof: RingPedersenProof<E, H, M>,
 }
 
-/// Generates the parameters needed for the h1_h2_N_tilde_vec. These parameters
-/// can be seen as environment variables for each party that they agree on. In
-/// this case, each new party generates it's own DlogStatements and submits it's
-/// proofs
-fn generate_h1_h2_n_tilde() -> (BigInt, BigInt, BigInt, BigInt, BigInt) {
-    let (ek_tilde, dk_tilde) =
-        Paillier::keypair_with_modulus_size(crate::PAILLIER_KEY_SIZE).keys();
-    let one = BigInt::one();
-    let phi = (&dk_tilde.p - &one) * (&dk_tilde.q - &one);
-    let h1 = BigInt::sample_below(&ek_tilde.n);
-    let (mut xhi, mut xhi_inv) = loop {
-        let xhi_ = BigInt::sample_below(&phi);
-        match BigInt::mod_inv(&xhi_, &phi) {
-            Some(inv) => break (xhi_, inv),
-            None => continue,
-        }
-    };
-    let h2 = BigInt::mod_pow(&h1, &xhi, &ek_tilde.n);
-    xhi = BigInt::sub(&phi, &xhi);
-    xhi_inv = BigInt::sub(&phi, &xhi_inv);
-    (ek_tilde.n, h1, h2, xhi, xhi_inv)
-}
-
 /// Generates the DlogStatement and CompositeProofs using the parameters
 /// generated by [generate_h1_h2_n_tilde]
-fn generate_dlog_statement_proofs(
-) -> (DLogStatement, CompositeDLogProof, CompositeDLogProof) {
-    let (n_tilde, h1, h2, xhi, xhi_inv) = generate_h1_h2_n_tilde();
-
-    let dlog_statement_base_h1 = DLogStatement {
-        N: n_tilde.clone(),
-        g: h1.clone(),
-        ni: h2.clone(),
+fn generate_dlog_statement_proofs() -> (
+    CompositeDLogStatement,
+    CompositeDLogProof,
+    CompositeDLogProof,
+) {
+    let (n_tilde, h1, h2, xhi, xhi_inv, phi) = generate_h1_h2_N_tilde();
+
+    let dlog_statement_base_h1 = CompositeDLogStatement {
+        modulus: n_tilde.clone(),
+        base: h1.clone(),
+        value: h2.clone(),
+    };
+    let dlog_witness_base_h1 = CompositeDLogWitness {
+        exponent: xhi,
+        totient: phi.clone(),
     };
 
-    let dlog_statement_base_h2 = DLogStatement {
-        N: n_tilde,
-        g: h2,
-        ni: h1,
+    let dlog_statement_base_h2 = CompositeDLogStatement {
+        modulus: n_tilde,
+        base: h2,
+        value: h1,
+    };
+    let dlog_witness_base_h2 = CompositeDLogWitness {
+        exponent: xhi_inv,
+        totient: phi.clone(),
     };
 
-    let composite_dlog_proof_base_h1 =
-        CompositeDLogProof::prove(&dlog_statement_base_h1, &xhi);
-    let composite_dlog_proof_base_h2 =
-        CompositeDLogProof::prove(&dlog_statement_base_h2, &xhi_inv);
+    let composite_dlog_proof_base_h1 = CompositeDLogProof::prove(
+        &dlog_statement_base_h1,
+        &dlog_witness_base_h1,
+    )
+    .unwrap();
```

Comment:
Also would rather not use `unwrap`s whenever we can. Can we add error handling @davidsemakula ?

Reply:
oops, sure thing ... was meant to be temp for quick refactoring, will add proper error handling.

Reply:
@drewstone `unwrap`s related to this are now only used in tests.